plugins/Installation/ServerFilesGenerator.php

```diff
@@ -195,7 +201,7 @@ protected static function createWebConfigFiles()
       <requestFiltering>
         <denyUrlSequences>
           <add sequence=".php" />
-        </denyUrlSequences>' . ($directoryToProtect === '/plugins' ? $additionForPlugins : '') . '
+        </denyUrlSequences>' . (!is_numeric($directoryToProtect) ? $additions : '') . '
```

Comment:
key->value items in the array have string keys, without the pair the key is numeric, so we can use this to check if we have some additions or not.